build/update-match-schema.sh

```diff
@@ -0,0 +1,38 @@
+# The purpose of the script is to build a golang string constant containing the
+# YAML code for the Match CRD. This is needed to auto generate the
+# JSONSchemaProps for Match.
+# It will parse the YAML for the Match CRD, found in $CRD_FILE, and output to
+# $GO_FILE. For controller-gen to generate the CRD, we must include the metadata
+# and typemeta fields. Since we don't want these fields to exist on the real
+# Match CRD, we embed the Match type in a dummy type that has the
+# metadata/typemeta fields, and then parse out these unwanted fields.
+
+GO_FILE="./pkg/target/matchcrd_constant.go"
+SRC_FILE="./pkg/mutation/match/match_types.go"
+CRD_FILE="./config/crd/bases/match.gatekeeper.sh_matchcrd.yaml"
+
+cat << EOF > ${GO_FILE}
+package target
+
+// DO NOT MODIFY THIS FILE DIRECTLY!
+// This file is generated from $SRC_FILE via "make manifests".
+
+const matchYAML = \`
+EOF
+
+# Delete apiVersion block, adjust indentation to un-embed the match field, escape backticks
```

Comment:
Everything but escape backticks seems like it could be done more reliably at bootstrapping time in golang, esp. since start/end trimming require positional consistency.

Reply:
Updated to do the "un-embedding" in go code